lang/c++/impl/DataFile.cc

```diff
@@ -63,35 +64,34 @@ boost::iostreams::zlib_params get_zlib_params() {
 }
 } // namespace
 
-DataFileWriterBase::DataFileWriterBase(const char *filename, const ValidSchema &schema, size_t syncInterval,
-                                       Codec codec) : filename_(filename),
-                                                      schema_(schema),
+DataFileWriterBase::DataFileWriterBase(const char *filename,
+                                       const ValidSchema &schema,
+                                       size_t syncInterval,
+                                       Codec codec) : DataFileWriterBase(fileOutputStream(filename), schema, syncInterval, codec) {
+}
+
+DataFileWriterBase::DataFileWriterBase(OutputStream &outputStream,
```

Comment:
I might be missing something here, but can't this constructor make `DataFileWriterBase` outlive `OutputStream` that it uses?